src/EPR.Payment.Service.Common.Data/Repositories/RegistrationFees/ProducerFeesRepository.cs

```diff
@@ -0,0 +1,78 @@
+﻿using EPR.Payment.Service.Common.Constants.RegistrationFees;
+using EPR.Payment.Service.Common.Data.Interfaces;
+using EPR.Payment.Service.Common.Data.Interfaces.Repositories.RegistrationFees;
+using Microsoft.EntityFrameworkCore;
+
+namespace EPR.Payment.Service.Common.Data.Repositories.RegistrationFees
+{
+    public class ProducerFeesRepository : IProducerFeesRepository
+    {
+        private readonly IAppDbContext _dataContext;
+
+        public ProducerFeesRepository(IAppDbContext dataContext)
+        {
+            _dataContext = dataContext;
+        }
+
+        public async Task<decimal> GetBaseFeeAsync(string producerType, string regulator, CancellationToken cancellationToken)
+        {
+            var currentDate = DateTime.UtcNow;
+
+            var fee = await _dataContext.RegistrationFees
+                .Where(r => r.Group.Type == GroupTypeConstants.ProducerType
+                            && r.SubGroup.Description.ToLower() == producerType.ToLower()
+                            && r.Regulator.Description == regulator
+                            && r.EffectiveFrom <= currentDate
+                            && r.EffectiveTo >= currentDate)
+                .OrderByDescending(r => r.EffectiveFrom)  // Ensure the most recent EffectiveFrom is selected
+                .Select(r => r.Amount)
+                .FirstOrDefaultAsync(cancellationToken);
+
+            if (fee == 0)
+                throw new KeyNotFoundException(string.Format(ProducerFeesRepositoryConstants.InvalidProducerTypeOrRegulatorError, producerType, regulator));
+
+            return fee;
+        }
+
+        public async Task<decimal> GetFirst20SubsidiariesFeeAsync(string regulator, CancellationToken cancellationToken)
+        {
+            var currentDate = DateTime.UtcNow;
+
+            var fee = await _dataContext.RegistrationFees
+                .Where(r => r.Group.Type == GroupTypeConstants.ProducerSubsidiaries
```

Comment:
same comment as above. we should not be using Description for filtering/joining the records

Reply:
Sure will change